source/distributor/service_impl.h

```diff
@@ -0,0 +1,54 @@
+#pragma once
+#ifdef __clang__
+#pragma clang diagnostic push
+#pragma clang diagnostic warning "-Wunused-parameter"
```

Comment:
Can we include a comment explaining why this is needed?

Reply:
Even better, trying it out, we no longer need this. I looks like one of the dep updates eliminated the need for this, si I removed the pragmas in f3bc7a7aac0579bd9be72d9cde1928fdf3453eba